sdk/messaging/azservicebus/sender.go

```diff
@@ -96,14 +100,42 @@ type ScheduleMessagesOptions struct {
 // delivered can be cancelled using `Receiver.CancelScheduleMessage(s)`
 // If the operation fails it can return an *azservicebus.Error type if the failure is actionable.
 func (s *Sender) ScheduleMessages(ctx context.Context, messages []*Message, scheduledEnqueueTime time.Time, options *ScheduleMessagesOptions) ([]int64, error) {
+	return scheduleMessages(ctx, s.links, s.retryOptions, messages, scheduledEnqueueTime)
+}
+
+// ScheduleAMQPMessagesOptions contains optional parameters for the ScheduleAMQPMessages function.
+type ScheduleAMQPMessagesOptions struct {
+	// For future expansion
+}
+
+// ScheduleAMQPMessages schedules a slice of Messages to appear on Service Bus Queue/Subscription at a later time.
+// Returns the sequence numbers of the messages that were scheduled.  Messages that haven't been
+// delivered can be cancelled using `Receiver.CancelScheduleMessage(s)`
+// If the operation fails it can return an *azservicebus.Error type if the failure is actionable.
+func (s *Sender) ScheduleAMQPMessages(ctx context.Context, messages []*AMQPMessage, scheduledEnqueueTime time.Time, options *ScheduleAMQPMessagesOptions) ([]int64, error) {
```

Comment:
Presumably other languages use overloads to avoid introducing additional methods.  What made you chose this over having `ScheduleMessage()` take an interface instead?

Reply:
Technically that would be a breaking change, so we're stuck.